app/models/mixins/retirement_mixin.rb

```diff
@@ -253,8 +253,8 @@ def valid_zone?
   end
 
   def system_context_requester
-    if try(:evm_owner_id).present?
-      User.find(evm_owner_id)
+    if try(:evm_owner_id).present? && User.find_by(:id => evm_owner_id)
+      evm_owner
     else
       $log.info("System context defaulting to admin user because owner of #{name} not set.")
```

Comment:
@d-m-u do you want to log the fact that there are orphaned resources without a parent. That message would help in debugging customer issues.

Reply:
see https://github.com/ManageIQ/manageiq/pull/18483#issuecomment-467188148

Reply:
@mkanoor @d-m-u and I discussed this and will add the logging in a new PR.